### PR TITLE
update to a modern theme with search

### PR DESCRIPTION
**Pull request by noklam**
## Description
Why was this PR created?

## Development notes
What have you changed, and how has this been tested?

## Checklist

- [ ] Read the [contributing](../CONTRIBUTING.md) guidelines
- [ ] Open this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Update the documentation to reflect the code changes
- [ ] Add a description of this change and add your name to the list of supporting contributions in the [`CHANGELOG.md`](../CHANGELOG.md) file. Please respect [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines.
- [ ] Add tests to cover your changes

## Notice

- [ ] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -47,6 +47,7 @@
     "sphinx_copybutton",
     "sphinx_markdown_tables",
     "myst_parser",
+    "sphinxawesome_theme.docsearch",
```
[the remark this answers is not in the document]
Comment (Galileo-Galilei) [reply docs/conf.py:50]: I'd be glad to have a better theme, but the CSS rendering is completely messed up for me too.  No hurry for this.

**Comment by noklam** [comment]
I think I somewhat make it works now, it seems that later version of sphinx breaking many themes. I have pin it <7.2.0 to fix this issue for now.

**Comment by noklam** [comment]
It build successfully but it doesn't look like exactly what I expected. I expected the theme to be like https://sphinxawesome.xyz/

**Comment by noklam** [comment]
Alright I found that I have missed an extension, the theme looks much nicer now. We can also setup Algolia for DocSearch, which is free for open source

**Pull request merged**